fastplotlib/layouts/_gridplot.py

```diff
@@ -25,128 +25,197 @@ def to_array(a) -> np.ndarray:
-                )
             # create the array representing the views for each subplot in the grid
             cameras = np.array([cameras] * self.shape[0] * self.shape[1]).reshape(
                 self.shape
             )
 
-        if isinstance(controllers, str):
-            if controllers == "sync":
-                controllers = np.zeros(
-                    self.shape[0] * self.shape[1], dtype=int
-                ).reshape(self.shape)
+        # list -> array if necessary
+        cameras = to_array(cameras).reshape(self.shape)
 
-        if controllers is None:
-            controllers = np.arange(self.shape[0] * self.shape[1]).reshape(self.shape)
+        if cameras.shape != self.shape:
+            raise ValueError("Number of cameras does not match the number of subplots")
+
+        # create the cameras
+        self._cameras = np.empty(self.shape, dtype=object)
+        for i, j in product(range(self.shape[0]), range(self.shape[1])):
+            self._cameras[i, j] = create_camera(camera_type=cameras[i, j])
+
+        if controller_ids is None:
+            # individual controller for each subplot
+            controller_ids = np.arange(self.shape[0] * self.shape[1]).reshape(self.shape)
+
+        elif isinstance(controller_ids, str):
+            if controller_ids == "sync":
+                controller_ids = np.zeros(self.shape, dtype=int)
+
+        # list controller_ids
+        elif isinstance(controller_ids, (list, np.ndarray)):
+            flat = list(chain(*controller_ids))
+
+            # list of str of subplot names, convert this to integer ids
+            if all([isinstance(item, str) for item in flat]):
+                if self.names is None:
+                    raise ValueError("must specify subplot `names` to use list of str for `controller_ids`")
+
+                # make sure each controller_id str is a subplot name
+                if not all([n in self.names for n in flat]):
+                    raise KeyError(
+                        f"all `controller_ids` strings must be one of the subplot names"
+                    )
 
-        controllers = to_array(controllers)
+                if len(flat) > len(set(flat)):
+                    raise ValueError(
+                        "id strings must not appear twice in `controller_ids`"
+                    )
 
-        if controllers.shape != self.shape:
-            raise ValueError
+                # initialize controller_ids array
+                ids_init = np.arange(self.shape[0] * self.shape[1]).reshape(self.shape)
 
-        cameras = to_array(cameras)
+                # set id based on subplot position for each synced sublist
+                for i, sublist in enumerate(controller_ids):
+                    for name in sublist:
+                        ids_init[self.names == name] = -(i + 1)  # use negative numbers because why not
 
-        self._controllers = np.empty(shape=cameras.shape, dtype=object)
+                controller_ids = ids_init
```

Comment:
should `controller_ids` be user-facing as a read only attr?

Reply:
I was thinking about this. I'm undecided whether Gridplots should allow these to be accessible, because then the user could try to change one of these controllers in the object array, but then it won't change the corresponding controller_id in the subplot(s). Right now the architecture is that the subplots _own_ the controllers and camera. Gridplot is mostly a way to create subplots. I was thinking of leaving the decision for when we make more comprehensive/flexible gridplots.